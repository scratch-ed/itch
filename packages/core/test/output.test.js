import { ResultManager } from '../src/output/index.ts';

test('normal usage', () => {
  const results = [];
  const manager = new ResultManager((r) => results.push(r));

  manager.startJudgement();
  manager.startGroup('Groep 1', 'show', 'test-sprite');
  manager.startTest('Hallo one test...');
  manager.appendMessage('Hallo');
  manager.appendDiff('expected', 'actual');
  manager.closeTest('wrong', 'Wrong');
  manager.closeGroup('A nice summary');
  manager.closeJudgement();

  expect(results).toMatchSnapshot();
});
