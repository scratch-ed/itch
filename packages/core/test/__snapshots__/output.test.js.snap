// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normal usage 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 1",
    "sprite": undefined,
    "tags": "test-sprite",
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Hallo one test...",
    "tags": [],
  },
  {
    "command": "append-message",
    "message": "Hallo",
  },
  {
    "actual": "actual",
    "command": "append-diff",
    "expected": "expected",
  },
  {
    "command": "close-test",
    "feedback": "Wrong",
    "status": "wrong",
  },
  {
    "command": "close-group",
    "summary": "A nice summary",
  },
  {
    "command": "close-judgement",
  },
]
`;
